PlaceRouteHierFlow/router/GcellDetailRouter.cpp

```diff
@@ -210,43 +210,24 @@ void GcellDetailRouter::Grid_Inactive_new(Grid &grid, std::set<RouterDB::SinkDat
       int mIdx = mit->metalIdx;
       if (drc_info.Metal_info[mIdx].direct == 0) {
         // vertical
-        if (mIdx < this->layerNo - 1) {
-          int vIdx = mIdx;
+        if (mIdx < this->layerNo - 1 || mIdx > 0) {
           box.LL.x = mit->coord[0].x;
           box.LL.y = mit->coord[0].y - drc_info.Metal_info[mIdx].dist_ee;
           box.UR.x = mit->coord[1].x;
           box.UR.y = mit->coord[1].y + drc_info.Metal_info[mIdx].dist_ee;
           // current metal cannot go up
           ConvertRect2GridPoints(plist, mIdx, box.LL.x, box.LL.y, box.UR.x, box.UR.y);
         };
-        if (mIdx > 0) {
-          int vIdx = mIdx - 1;
-          box.LL.x = mit->coord[0].x;
-          box.LL.y = mit->coord[0].y - drc_info.Metal_info[mIdx].dist_ee;
-          box.UR.x = mit->coord[1].x;
-          box.UR.y = mit->coord[1].y + drc_info.Metal_info[mIdx].dist_ee;
-          ConvertRect2GridPoints(plist, mIdx, box.LL.x, box.LL.y, box.UR.x, box.UR.y);
-        };
```

Comment:
duplicated code here, some points are added to plist twice, causing set insert slow